comitup/config.py

```diff
@@ -43,29 +54,82 @@ def __getattr__(self, tag):
             raise AttributeError
 
 
+def _getserial():
+    # Extract serial from cpuinfo file
+    cpuserial = "0000000000000000"
+    try:
+        with open('/proc/cpuinfo', 'r') as f:
+            for line in f:
+                if line.startswith('Serial'):  # Serial		: 10000000082acda8
+                    cpuserial = line.split(':')[-1].strip()
+                    break
+
+    except Exception as e:
+        log.error("Failed to parse /proc/cpuinfo for serial number")
+        log.error(e)
+        cpuserial = "ERROR000000000"
+
+    return cpuserial
+
+
+def _get_mac():
+    mac = "000000000000"
+    cmd = "nmcli -f GENERAL.HWADDR device show wlan0".split(' ')
+    try:
+        mac_info = subprocess.check_output(cmd).decode("utf-8")
+        # from: GENERAL.HWADDR:                         DC:A6:32:2B:6D:37
+        # to  : DCA6322B6D37
+        mac = mac_info.replace('GENERAL.HWADDR:', '').replace(':', '').strip()
+
+    except Exception as e:
+        log.error(e)
+
+    return mac
+
+
 def load_data():
     if os.path.isfile(BOOT_CONF_PATH):
         try:
             dest = shutil.copyfile(BOOT_CONF_PATH, CONF_PATH)
             print("Boot config file copied:", dest)
+            log.info("Boot config file copied: {}".format(dest))
             os.remove(BOOT_CONF_PATH)
         except:
             print("Error occurred while copying file.")
+            log.error("Error occurred while copying file.")
 
     conf = Config(
                 CONF_PATH,
                 defaults={
-                    'ap_name': 'comitup-<nnn>',
+                    'ap_name': DEFAULT_AP,
                     'ap_password': '',
                     'web_service': '',
                     'service_name': 'comitup',
                     'external_callback': '/usr/local/bin/comitup-callback',
                 },
              )
 
+    # validate that if the ap_name has '<' or '>' and '-', it follows spec
+    if ('<' in conf.ap_name or '>' in conf.ap_name) and \
```

Comment:
Again - unnecessarily restrictive.

Reply:
agree.  I fretted over if a typo is made in `comitup.conf` and how the code would deal with it.  
I hoped that this code would only be on the path that sees the ap_name from `comitup.conf`.